src/main/java/io/confluent/connect/jdbc/util/CachedConnectionProvider.java

```diff
@@ -76,7 +76,12 @@ public boolean isConnectionValid(
       Connection connection,
       int timeout
   ) throws SQLException {
-    return provider.isConnectionValid(connection, timeout);
+    try {
+      return provider.isConnectionValid(connection, timeout);
+    } catch (SQLException sqle) {
+      log.error("Error trying to check if the underlying connection is valid", sqle);
```

Comment:
Hi @gharris1727 , thank you for your review and suggestions.

I agree with you, it's better if we don't give the wrong impression that there is some irrecoverable condition.

I also changed the target branch of this PR, but honestly it seems that my branch was too behind the 5.0.x branch and a lot has changed since I created the PR.
Maybe it will be easier if I create another PR from the correct branch, or do you think that merging this PR won't cause any major trouble?
Thank you again!

Regards

Reply:
@Dudssource Thanks for updating this PR.

You don't have to create a new PR, we can fix up this one. If you change your branch locally and then force-push the branch to your fork, and it will automatically update this PR.

You can rebase your changes on the right base branch with either `git rebase -i 5.0.x` or by cherry-picking each commit separately. You can discard all of the commits that you didn't make, since they came from the upstream/master branch, and we don't want them in this PR.

Unfortunately because you're merging from your master branch, the step above messes up your master branch temporarily. Once we're done with this PR, you can force-update your fork's master to be even with upstream/master again, and everything will be back to normal.